src/counters.rs

```diff
@@ -0,0 +1,178 @@
+        counter_type: CounterType,
+    ) -> CreateCounterReq<'i> {
+        CreateCounterReq {
+            infinispan_client,
+            name: name.into(),
+            counter_type,
+            value: None,
+        }
+    }
+
+    pub fn with_value(mut self, value: CounterVal) -> CreateCounterReq<'i> {
+        self.value = Some(value);
+        self
+    }
+
+    pub async fn send(&self) -> Result<Response, reqwest::Error> {
+        let counter = match self.counter_type {
+            CounterType::Weak => Counter::Weak(WeakCounter {
+                initial_value: self.value,
+            }),
+            CounterType::Strong => Counter::Strong(StrongCounter {
+                initial_value: self.value,
+            }),
+        };
+
+        let http_request = Request::post(self.infinispan_client.counter_url(&self.name))
+            .header(
+                AUTHORIZATION,
+                &self.infinispan_client.basic_auth_encoded_val,
+            )
+            .body(json!(counter).to_string())
+            .unwrap();
+
+        InfinispanRequest::new(http_request, self.infinispan_client.http_client.clone())
+            .send()
+            .await
+    }
+}
+
+pub struct IncrementCounterReq<'i> {
+    infinispan_client: &'i Infinispan,
+    name: String,
+    delta: Option<CounterVal>,
+}
+
+impl<'i> IncrementCounterReq<'i> {
+    pub fn new(
+        infinispan_client: &'i Infinispan,
+        name: impl Into<String>,
+    ) -> IncrementCounterReq<'i> {
+        IncrementCounterReq {
+            infinispan_client,
+            name: name.into(),
+            delta: None,
+        }
+    }
+
+    pub fn by(mut self, delta: CounterVal) -> IncrementCounterReq<'i> {
+        self.delta = Some(delta);
+        self
+    }
+
+    pub async fn send(&self) -> Result<Response, reqwest::Error> {
+        let http_request = Request::post(self.query_with_args())
+            .header(
+                AUTHORIZATION,
+                &self.infinispan_client.basic_auth_encoded_val,
+            )
+            .body("".into())
+            .unwrap();
+
+        InfinispanRequest::new(http_request, self.infinispan_client.http_client.clone())
+            .send()
+            .await
+    }
+
+    fn query_with_args(&self) -> String {
+        let mut query = format!(
+            "{}?action=add",
+            self.infinispan_client.counter_url(&self.name)
+        );
+
+        if let Some(delta) = self.delta {
+            query.push_str(&format!("&delta={}", delta));
+        }
+
+        query
+    }
+}
+
+impl Infinispan {
+    pub fn create_weak_counter(&self, name: impl Into<String>) -> CreateCounterReq {
+        CreateCounterReq::new(&self, name, CounterType::Weak)
+    }
+
+    pub fn create_strong_counter(&self, name: impl Into<String>) -> CreateCounterReq {
+        CreateCounterReq::new(&self, name, CounterType::Strong)
+    }
+
+    pub fn get_counter_value(&self, name: impl AsRef<str>) -> InfinispanRequest {
```

Comment:
Same here as below - return a Result in your public API, hard to avoid it if you don't want panics...